src/Perpetuum/Services/EventServices/EventProcessors/WeatherWatcher.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Perpetuum.ExportedTypes;
+using Perpetuum.Services.EventServices.EventMessages;
+using Perpetuum.Services.Weather;
+using Perpetuum.Zones;
+using Perpetuum.Zones.Effects.ZoneEffects;
+using System;
+
+namespace Perpetuum.Services.EventServices.EventProcessors
+{
+    public class WeatherWatcher : EventProcessor<EventMessage>
+    {
+        private IZone _zone;
+        private readonly Lazy<ZoneEffect> _goodWeather;
+        private readonly Lazy<ZoneEffect> _badWeather;
+        public WeatherWatcher(IZone zone)
+        {
+            _zone = zone;
+            _goodWeather = new Lazy<ZoneEffect>(GetGoodWeather);
+            _badWeather = new Lazy<ZoneEffect>(GetBadWeather);
+        }
+
+        private ZoneEffect GetGoodWeather()
```

Comment:
Perhaps `CreateGoodWeather` would be a little more clear? Same for Bad Weather.

Reply:
Same as previous, this *is* a `Get` operation. Title is correct.